packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1867,8 +1868,11 @@ export function handleAllTypes(context: TCGCContext): [void, readonly Diagnostic
     }
   }
   // update for orphan models/enums/unions
-  for (const client of listClients(context)) {
-    const namespaces = [client.service];
+  const allNamespaces = [...context.program.getGlobalNamespaceType().namespaces.values()].filter(
```

Comment:
when generating name for anonymous model, there is a similar logic. see here: https://github.com/Azure/typespec-azure/blob/70dab25c415321e29f251a4ee42427ef13264ce8/packages/typespec-client-generator-core/src/public-utils.ts#L349.